hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -480,31 +480,32 @@ Scramble enablement is done differently depending on the type of partitions.
 *  For information partitions, the scramble enablement is done on a per page basis.
    *  Software can configure for each page whether scramble is enabled.
 
-#### Flash ECC
+#### Flash ECC and ICV
 
-There are two types of flash ECC supported.
+Flash supports both ECC (error correction) and ICV (integrity check value).
+While the two are used for different functions, they are implemented as two separate ECCs, thus flash supports two types of ECC.
 
-The first type is an integrity ECC used to detect whether the de-scrambled data has been modified.
-The second type is a reliability ECC used for error detection and correction on the whole flash word.
+ICV is an integrity check, implemented as ECC, used to detect whether the de-scrambled data has been modified.
+The other is a reliability ECC used for error detection and correction on the whole flash word.
 
-The first type of ECC is required on every flash word.
-The second type of ECC is configurable based on the various page and memory property configurations.
+The key differentiation here is that ICV is used only for detection, while the real error correction can correct single bit errors.
+Both ICV and ECC are configurable based on the various page and memory property configurations.
 
-##### Overall ECC Application
+##### Overall ICV and ECC Application
 
-The following diagram shows how the various ECC tags are applied and used through the life of a transactions.
+The following diagram shows how the various ICV / ECC tags are applied and used through the life of a transactions.
 ![Flash ECC_LIFE](flash_integrity.svg).
 
-Note that the integrity ECC is calculated over the descrambled data and is only 4-bits.
+Note that the ICV (integrity ECC) is calculated over the descrambled data and is only 4-bits.
 While the reliability ECC is calculated over both the scrambled data and the integrity ECC.
 
-##### Integrity ECC
+##### ICV
 
-The purpose of the integrity ECC is to emulate end-to-end integrity like the other memories.
+The purpose of the ICV (integrity check value, implemnted as an ECC) is to emulate end-to-end integrity like the other memories.
```

Comment:
```suggestion
The purpose of the ICV (integrity check value, implemented as ECC) is to emulate end-to-end integrity like the other memories.
```

Reply:
hmmm... this seems subjective to me. I'm not going to change this one.

Reply:
But please change implemnted -> implemented